core/parse-xml.c

```diff
@@ -1999,12 +2078,22 @@ int parse_dlf_buffer(unsigned char *buffer, size_t size, struct dive_table *tabl
 			case 3:
 				/* Measure Oxygen */
 				//printf("%d s: o2 cells(0.01 mV): %d %d %d %d\n", time, (ptr[5] << 8) + ptr[4], (ptr[7] << 8) + ptr[6], (ptr[9] << 8) + ptr[8], (ptr[11] << 8) + ptr[10]);
+				// [Pa/mV] coeficient O2
+				// 100 Pa == 1 mbar
+				sample_start();
+				cur_sample->time.seconds = time;
+				cur_sample->o2sensor[0].mbar = ( ((ptr[5] << 8) + ptr[4]) * o2_sensor_calibration_values[0]) / 10000;
+				cur_sample->o2sensor[1].mbar = ( ((ptr[7] << 8) + ptr[6]) * o2_sensor_calibration_values[1]) / 10000;
+				cur_sample->o2sensor[2].mbar = ( ((ptr[9] << 8) + ptr[8]) * o2_sensor_calibration_values[2]) / 10000;
+				// Subsurface only handles 3 o2 sensors.
+				//cur_sample->o2sensor[3].mbar = ( ((ptr[11] << 8) + ptr[10]) * o2_sensor_calibration_values[3]) / 10000;
+				sample_end();
 				break;
 			case 4:
 				/* Measure GPS */
 				cur_latitude.udeg =  (int)((ptr[7]  << 24) + (ptr[6]  << 16) + (ptr[5] << 8) + (ptr[4] << 0));
 				cur_longitude.udeg = (int)((ptr[11] << 24) + (ptr[10] << 16) + (ptr[9] << 8) + (ptr[8] << 0));
-				cur_dive->dive_site_uuid = create_dive_site_with_gps(NULL, cur_latitude, cur_longitude, cur_dive->when);
+				cur_dive->dive_site_uuid = create_dive_site_with_gps("DLF imported", cur_latitude, cur_longitude, cur_dive->when);
```

Comment:
Then we end up with multiple different sites named the same thing. It would probably be better if the ui handled divesites without a name better, but I felt that was out of scope.